lib/pure/typetraits.nim

```diff
@@ -107,6 +107,28 @@ since (1, 3, 5):
 
 import std/macros
 
+macro len*(T: typedesc[enum]): int =
+  ## Returns the number of items in the enumeration `T`.
+
+  runnableExamples:
+    type Foo = enum fooItem1 fooItem2
+    doAssert Foo.len == 2
+
+  var impl = getTypeImpl(T)
```

Comment:
fails for `type Foo[T] = enum k0 k1 k2`
`echo Foo[float].len`

this seems to work and is simpler:

```nim
  macro len(T: typedesc[enum]): int =
    var impl = getType(T)
    assert impl.kind == nnkBracketExpr
    impl = impl[1]
    assert impl.kind == nnkEnumTy
    newLit(impl.len - 1)
```

Reply:
Yes, it works. I did it this way and I also added an additional check to the test code to ensure that it works for a type alias chain.